bot_file.py

```diff
@@ -172,8 +174,15 @@ def when_you_hear(self, channel: str, pattern: str) -> None:
         """ notify the user when you see a pattern """
         person = self._last_sender
 
+        try:
+            re.compile(pattern)
```

Comment:
re: detecting `.` / `.*`

```python
>>> import sre_parse
>>> sre_parse.parse('..*.{0,}')
[('any', None), ('max_repeat', (0, 4294967295, [('any', None)])), ('max_repeat', (0, 4294967295, [('any', None)]))]
```

could be useful. hmm, or keep a message history of 1000 or so and refuse to register a pattern that's going to be too noisy?

and/or deliver pings to the `@slackbot` channel.